packages/editor/src/client/panel/imageEditPanel.tsx

```diff
@@ -193,6 +195,62 @@ export function ImagedEditPanel({id, file, onClose, onSave}: ImageEditPanelProps
     }
   }
 
+  /**
+   * Resizes an image on client side, if larger than 10 MB, which is the upload max of the api
+   * @param file
+   */
+  async function resizeImage(file: File): Promise<File> {
+    // only resize image if larger than 10 MB
+    if (!willImageResize(file)) {
+      return file // do not resize
+    }
+    const originalFileName: string = file.name
+    if (!originalFileName) {
```

Comment:
I would avoid throwing errors here. I'm not sure if it is even possible for a file not to have a name, but if there is no name i would just create a random name e.g. "unnamed_image_%timestamp%". You might need to get the right file extension for the name.

Reply:
ok